opcua/ua/uatypes.py

```diff
@@ -163,12 +163,35 @@ class EventNotifier(_MaskEnum):
 
 class Guid(FrozenClass):
 
-    def __init__(self):
-        self.uuid = uuid.uuid4()
+    def __init__(self, value=None):
+        if value is None or value == "":
+            # if no string is supplied, create a new UUID
+            self.uuid = uuid.uuid4()
+        else:
+            try:
+                # attempt to make a UUID with the supplied value, if hex string has errors use a default string
+                self.uuid = uuid.UUID(value)
+            except ValueError as e:
+                # TODO should notify user of error, but this module has no logger, raise UaError stops execution
+                self.uuid = uuid.UUID("00000000-0000-0000-0000-000000000000")
+
         self._freeze = True
 
     def to_binary(self):
-        return self.uuid.bytes
+        # NOTE THAT self.uuid.bytes WILL NOT WORK FOR UA BINARY PROTOCOL; BYTE ORDER IS NOT CORRECT
+        # convert python UUID 6 field format to OPC UA 4 field format
+        guid = self.uuid
+        f1 = uabin.Primitives.UInt32.pack(guid.fields[0])
+        f2 = uabin.Primitives.UInt16.pack(guid.fields[1])
+        f3 = uabin.Primitives.UInt16.pack(guid.fields[2])
+        f4a = uabin.Primitives.Byte.pack(guid.fields[3])
+        f4b = uabin.Primitives.Byte.pack(guid.fields[4])
+        f4c = bytearray(struct.unpack('8B', struct.pack('>Q', guid.fields[5]))[2:8])  # no primitive .pack available
```

Comment:
It don't think it will work. I basically cheated here. uuid.fields[5] is actually a 6 byte int. That is why I pack it to bytes and disregard the unused bytes.